tracing-subscriber/src/registry/mod.rs

```diff
@@ -334,18 +389,50 @@ where
     /// Returns the ID of this span's parent, or `None` if this span is the root
     /// of its trace tree.
     pub fn parent_id(&self) -> Option<&Id> {
```

Comment:
I would probably deprecate this method, make _large_ note of how this doesn't work with per-layer filtering, and make a method that actually works.

Reply:
the other option is to just add a big comment that says "this is a lower-level API, it doesn't take filtering into account blah blah blah". i feel like there _might_ be cases where it's needed, but idk.

Reply:
honestly --- and i will admit this is a bit silly --- my biggest concern about deprecating it might just be "what would we call the replacement function?"

i can't really come up with a better name for this than `parent_id`, everything else seems strictly worse...`get_parent_id`? ew...

Reply:
hmm, that's true. those names are kinda crappy. eh, i'm okay with a big warning in that case—it _is_ a massive feature landing in a point release and the impact of this slightly cludgy API is scoped to layer implementers.

Reply:
Could we deprecate it and suggest `.parent().map(|p| p.id())` instead? It's not beautiful, but it works, and it means we don't have to come up with an alternative name.

Reply:
90ac30ccd5619535d7f1fae9b04f25bf56ca21f5 deprecates this.